repos/fdbt-site/src/pages/api/managePurchaseMethod.ts

```diff
@@ -86,12 +93,59 @@ export default async (req: NextApiRequestWithSession, res: NextApiResponse): Pro
                     id: 'purchase-method-name',
                 });
             }
+
+            if (id) {
+                const purchaseDetails = await getSalesOfferPackageByIdAndNoc(id, noc);
+                if (purchaseDetails.isCapped) {
+                    const dbIsCapped = purchaseDetails.isCapped;
+
+                    const validPurchaseLocations = purchaseLocationsList(dbIsCapped).method;
+                    const isValidPurchaseLocation = toArray(purchaseLocations).every((val) =>
+                        validPurchaseLocations.includes(val),
+                    );
+
+                    if (!isValidPurchaseLocation) {
+                        errors.push({
+                            errorMessage: `Select the valid option(s) for the purchase locations.`,
+                            id: purchaseLocationsList(dbIsCapped).id,
+                        });
+                    }
+
+                    const validPaymentMethods = paymentMethodsList(dbIsCapped).paymentMethods;
+                    const isValidPaymentMethod = toArray(paymentMethods).every((val) =>
+                        validPaymentMethods.includes(val),
+                    );
+
+                    if (!isValidPaymentMethod) {
+                        errors.push({
+                            errorMessage: `Select the valid option(s) for the payment method.`,
+                            id: paymentMethodsList(dbIsCapped).id,
+                        });
+                    }
+
+                    const validTicketFormats = ticketFormatsList(dbIsCapped).ticketFormats.map(
+                        (ticketFormat) => ticketFormat.value,
+                    );
+                    const isValidTicketFormat = toArray(ticketFormats).every((val) => validTicketFormats.includes(val));
```

Comment:
can we not use 'val' as its difficult to understand what we're iterating over

Reply:
ticketFormat contains {display: "", value: ""}

Reply:
so we should be calling them 'ticketFormatValue' - val is not an appropriate name for a variable.